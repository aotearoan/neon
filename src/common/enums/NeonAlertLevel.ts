/**
 * Alert levels are the severity levels to be used with <a href="/feedback/alert#examples">NeonAlert & NeonToast</a>
 * components to inform the user of the severity of a message.
 */
export enum NeonAlertLevel {
  /** Information level, use for standard informational messages. */
  Info = 'info',
  /** Success level, use for indicating successful actions. */
  Success = 'success',
  /** Warning level, use for indicating a problem that the user <strong>can</strong> recover from. */
  Warn = 'warn',
  /** Error level, use for indicating a problem that the user <strong>cannot</strong> recover from. */
  Error = 'error',
}
