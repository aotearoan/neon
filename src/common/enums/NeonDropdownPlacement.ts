/**
 * Describes placement of dropdown related components like:
 * <ul class="no-style">
 *  <li><a href="/user-input/select">NeonSelect</a></li>
 *  <li><a href="/navigation/dropdown-menu">NeonDropdownMenu</a></li>
 *  <li><a href="/presentation/dropdown">NeonDropdown</a></li>
 * </ul>
 *
 * <p>The placement has a <strong>primary placement</strong> (the first direction) & a <strong>secondary placement</strong>
 * (the second direction). The primary placement is the location of the popup relative to the triggering element (e.g.
 * the select button). With the secondary placement defining the alignment relative to the trigger. E.g. BottomLeft will
 * open the dropdown below the trigger element with the left-hand edges aligned.</p>
 *
 * <p>The placement utilities are smart & dynamic. They listen to resize events & aim for best effort based on where the
 * dropdown is opened. E.g. in the case above with BottomLeft if there is not enough visible room to align the component
 * to the left then it will try to align to the right & likewise if there is not enough space below the trigger it will
 * align to the top (& reverse the option order in the case of a select).</p>
 */
export enum NeonDropdownPlacement {
  /** Placement above trigger element, aligned to the left */
  TopLeft = 'top-left',
  /** Placement above trigger element, aligned to the right */
  TopRight = 'top-right',
  /** Placement below trigger element, aligned to the left */
  BottomLeft = 'bottom-left',
  /** Placement below trigger element, aligned to the right */
  BottomRight = 'bottom-right',
  /** Placement to the left of the trigger element, aligned to the top */
  LeftTop = 'left-top',
  /** Placement to the left of the trigger element, aligned to the bottom */
  LeftBottom = 'left-bottom',
  /** Placement to the right of the trigger element, aligned to the top */
  RightTop = 'right-top',
  /** Placement to the right of the trigger element, aligned to the bottom */
  RightBottom = 'right-bottom',
}
