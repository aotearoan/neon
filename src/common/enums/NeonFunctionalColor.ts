/**
 * The core color palettes available for styling components in Neon. See the design section
 * <a href="/design/color">Color</a> page for more details on color palettes.
 */
export enum NeonFunctionalColor {
  /** Low contrast is a more subtle neutral palette which can be used for secondary actions & supporting text. */
  LowContrast = 'low-contrast',
  /** Neutral is the base greyscale palette. */
  Neutral = 'neutral',
  /**
   * High contrast is a higher contrast palette than can be used for increased accessibility or for a stronger black &
   * white application.
   * */
  HighContrast = 'high-contrast',
  /** Brand is the main brand color palette, this is useful for branded CTAs, text & controls. */
  Brand = 'brand',
  /** Primary is the main application color palette. For most applications this will be used by default. */
  Primary = 'primary',
  /** Info can be used to convey informational context, e.g. notes & alerts */
  Info = 'info',
  /** Success can be used to convey successful actions. */
  Success = 'success',
  /** Warn can be used to convey issues/actions that are less severe but may still impact the user. */
  Warn = 'warn',
  /** Error can be used to convey issues/actions that are severe & that the user may not be able to recover from. */
  Error = 'error',
}
