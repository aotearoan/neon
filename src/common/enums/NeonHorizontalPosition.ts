/**
 * Describes the horizontal positioning of certain elements in the context of a component. An example would be an icon
 * in a button or label as well as the label for a checkbox or switch.
 */
export enum NeonHorizontalPosition {
  /** Left horizontal positioning. */
  Left = 'left',
  /** Right horizontal positioning. */
  Right = 'right',
}
