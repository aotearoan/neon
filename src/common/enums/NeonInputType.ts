/**
 * Corresponds to the <a href="https://developer.mozilla.org/en-US/docs/Web/HTML/Element/input#input_types">HTML input
 * types</a> for use with <a href="/user-input/input">NeonInput</a>.
 *
 * NOTE: prefer the specific components for Number, Password, Date, File, Search fields.
 */
export enum NeonInputType {
  /** Date input for selecting dates from a calendar. Use <a href="/user-input/date-picker">NeonDatePicker</a>. */
  Date = 'date',
  /** Date time input for selecting dates including a specific time. */
  DateTime = 'datetime-local',
  /** Email input for entering email addresses. */
  Email = 'email',
  /**
   * File input for opening a file dialog. Use <a href="/user-input/file">NeonDatePicker</a> or
   * <a href="/user-input/drop-zone">NeonDropZone</a>.
   */
  File = 'file',
  /** Number input for entering numeric values. Use <a href="/user-input/number">NeonNumber</a> */
  Number = 'number',
  /** Password input for entering passwords. Use <a href="/user-input/password">NeonPassword</a> */
  Password = 'password',
  /**
   * Range input for a slider control. Use <a href="/user-input/slider">NeonSlider</a> or
   * <a href="/user-input/range-slider">NeonRangeSlider</a>
   */
  Range = 'range',
  /** Search input for searching. Use <a href="/user-input/search">NeonSearch</a> */
  Search = 'search',
  /** Telephone number input. */
  Tel = 'tel',
  /** Text input for single or multiline text input. */
  Text = 'text',
  /** Time input. */
  Time = 'time',
  /** URL input for entering web addresses. */
  Url = 'url',
}
