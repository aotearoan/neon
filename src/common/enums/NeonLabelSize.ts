/**
 * The size options for the <a href="/presentation/label">NeonLabel</a> component.
 */
export enum NeonLabelSize {
  /** Extra extra small, default is 16 rem. */
  ExtraExtraSmall = 'xxs',
  /** Extra small, default is 20 rem. */
  ExtraSmall = 'xs',
  /** Small, default is 28 rem. */
  Small = 's',
  /** Medium, default is 36 rem. */
  Medium = 'm',
  /** Large, default is 44 rem. */
  Large = 'l',
}
