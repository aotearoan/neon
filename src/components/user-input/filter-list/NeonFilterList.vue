<template>
  <div
    :aria-activedescendant="
      multiple && Array.isArray(modelValue) ? modelValue[0] : (!Array.isArray(modelValue) && modelValue) || undefined
    "
    :aria-multiselectable="multiple"
    :class="[
      `neon-filter-list--${color}`,
      `neon-list--${size}`,
      `neon-filter-list--${size}`,
      { 'neon-filter-list--multiple': multiple },
    ]"
    class="neon-filter-list"
    role="listbox"
  >
    <div
      v-for="item in visibleItems"
      :key="item.key"
      :aria-selected="selected[item.key]"
      :class="[
        {
          'neon-filter-list__item--disabled': item.disabled,
          'neon-filter-list__item--enabled': !item.disabled,
          'neon-filter-list__item--selected': selected[item.key],
        },
      ]"
      :tabindex="!item.disabled ? 0 : -1"
      class="neon-list__item neon-filter-list__item"
      role="option"
      @click.stop.prevent="!item.disabled && toggleItem(item.key, $event)"
      @keydown.enter.stop.prevent="!item.disabled && toggleItem(item.key)"
      @keydown.space.stop.prevent="!item.disabled && toggleItem(item.key)"
    >
      <span class="neon-filter-list__item-label">{{ item.label }}</span>
      <span v-if="item.count" class="neon-filter-list__item-count">{{ item.count.toLocaleString() }}</span>
      <neon-icon
        v-if="!item.disabled && selected[item.key]"
        :color="color"
        :disabled="item.disabled"
        class="neon-filter-list__item-close"
        name="times"
      />
    </div>
    <neon-link v-if="displayShowAllToggle" class="neon-filter-list__show-toggle" @click="toggleShowAll()">
      {{ toggleShowAllLabel }}
    </neon-link>
  </div>
</template>

<script lang="ts" src="./NeonFilterList.ts"></script>
