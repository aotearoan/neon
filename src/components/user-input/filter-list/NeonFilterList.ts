import { computed, defineComponent, ref } from 'vue';
import { NeonSize } from '../../../common/enums/NeonSize';
import { NeonFunctionalColor } from '../../../common/enums/NeonFunctionalColor';
import NeonIcon from '../../presentation/icon/NeonIcon.vue';
import type { NeonFilterListItem } from '../../../common/models/NeonFilterListItem';
import NeonLink from '../../navigation/link/NeonLink.vue';

/**
 * <p><strong>NeonFilterList</strong> is an alternative component to a select where the values are displayed in a
 * visible list to the user. This component is particularly useful for presenting filters to the user and also provides
 * the option of displaying the item counts with each list item.</p>
 */
export default defineComponent({
  name: 'NeonFilterList',
  components: {
    NeonIcon,
    NeonLink,
  },
  props: {
    /**
     * The list if items to display
     */
    items: { type: Array as () => Array<NeonFilterListItem>, required: true },
    /**
     * Either a single string, indicating the key of the selected item or an array of selected keys in the case
     * multiple = true.
     */
    modelValue: { type: [String, Array as () => Array<string>], required: true },
    /**
     * Allow multi-select of items.
     */
    multiple: { type: Boolean, default: true },
    /**
     * The size of the list items - Small, Medium or Large.
     */
    size: { type: String as () => NeonSize, default: NeonSize.Medium },
    /**
     * The color of the selected list items..
     */
    color: { type: String as () => NeonFunctionalColor, default: NeonFunctionalColor.LowContrast },
    /**
     * The number of visible items. If there are more items they will be placed behind an expansion
     * button which toggles displaying displayCount items and all items.
     */
    displayCount: { type: Number, default: null },
    /**
     * Label for the Show more toggle if <em>displayCount</em> is set. The default is 'Show {count} more' where {count} is substituted with the display count
     */
    showMoreLabel: { type: String, default: 'Show {count} more' },
    /**
     * Label for the expanded Show more toggle if <em>displayCount</em> is set. The default is 'Show less'.
     */
    showLessLabel: { type: String, default: 'Show less' },
  },
  emits: [
    /**
     * emitted when the user selects or toggles the selection of an item.
     * @type {string | string[]} either the selected option's key (single select) or an array of the selected keys
     * (multi-select).
     */
    'update:modelValue',
  ],
  setup(props, { emit }) {
    const showAll = ref(false);

    const selected = computed((): Record<string, boolean> => {
      const result: Record<string, boolean> = {};
      (props.multiple ? (props.modelValue as string[]) : [props.modelValue as string]).forEach((v) => (result[v] = true));
      return result;
    });

    const displayShowAllToggle = computed(() => props.displayCount && props.items.length > props.displayCount);

    const visibleItems = computed(() => {
      return displayShowAllToggle.value && !showAll.value
        ? props.items.filter((item, index) => !props.displayCount || index < props.displayCount)
        : props.items;
    });

    const computedShowMoreLabel = computed(() => props.showMoreLabel.replace('{count}', `${props.items.length - visibleItems.value.length}`));

    const toggleShowAllLabel = computed(() => showAll.value ? props.showLessLabel : computedShowMoreLabel.value);

    const emitInput = (value: string | string[]) => {
      emit('update:modelValue', value);
    };

    const toggleItem = (key: string, $event?: Event) => {
      if (props.multiple) {
        if (selected.value[key]) {
          emitInput((props.modelValue as string[]).filter((v) => v !== key));
        } else {
          const result = [];
          (props.modelValue as string[]).forEach((v) => result.push(v));
          result.push(key);
          emitInput(result);
        }
      } else {
        emitInput(key === props.modelValue ? '' : key);
      }

      if ($event) {
        ($event.target as HTMLDivElement).blur();
      }
    };

    const toggleShowAll = () => {
      showAll.value = !showAll.value;
    };

    return {
      showAll,
      selected,
      visibleItems,
      toggleShowAllLabel,
      displayShowAllToggle,
      toggleItem,
      toggleShowAll,
    };
  },
});
