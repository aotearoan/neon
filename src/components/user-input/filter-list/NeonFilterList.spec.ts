import NeonFilterList from './NeonFilterList.vue';
import { NeonFunctionalColor } from '@/common/enums/NeonFunctionalColor';
import { NeonSize } from '@/common/enums/NeonSize';
import type { RenderResult } from '@testing-library/vue';
import { fireEvent, render } from '@testing-library/vue';
import { router } from '@/../test/unit/test-router';

describe('NeonFilterList', () => {
  const items = [
    {
      key: 'key1',
      label: 'Item 1',
      count: 5837,
    },
    {
      key: 'key2',
      label: 'Item 2',
      count: 433,
    },
    {
      key: 'key3',
      label: 'Item 3',
      count: 327,
    },
    {
      key: 'key4',
      label: 'Disabled item',
      count: 100,
      disabled: true,
    },
  ];

  const longItemList = [
    {
      key: 'key1',
      label: 'Item 1',
      count: 5837,
    },
    {
      key: 'key2',
      label: 'Item 2',
      count: 433,
    },
    {
      key: 'key3',
      label: 'Item 3',
      count: 327,
    },
    {
      key: 'key4',
      label: 'Item 4',
      count: 100,
    },
    {
      key: 'key5',
      label: 'Item 5',
      count: 5837,
    },
    {
      key: 'key6',
      label: 'Item 6',
      count: 433,
    },
    {
      key: 'key7',
      label: 'Item 7',
      count: 327,
    },
    {
      key: 'key8',
      label: 'Item 8',
      count: 100,
    },
  ];

  let harness: RenderResult;

  beforeEach(() => {
    harness = render(NeonFilterList, {
      props: { items, modelValue: [] },
      global: {
        plugins: [router],
      },
    });
  });

  it('renders multiselect short list', () => {
    const { container, html } = harness;
    expect(html()).toMatchSnapshot();
    expect(container.querySelectorAll('.neon-filter-list__item').length).toEqual(items.length);
  });

  it('renders long list displayCount', () => {
    const displayCount = 3;
    const { container } = render(NeonFilterList, {
      props: { items: longItemList, displayCount, modelValue: [] },
      global: {
        plugins: [router],
      },
    });
    expect(container.querySelectorAll('.neon-filter-list__item').length).toEqual(displayCount);
  });

  it('renders long list displayCount, showAll', async () => {
    const displayCount = 3;
    const { container } = render(NeonFilterList, {
      props: { items: longItemList, modelValue: [], displayCount },
      global: {
        plugins: [router],
      },
    });
    const el = container.querySelector('.neon-filter-list__show-toggle') as HTMLInputElement;
    await fireEvent.click(el);
    expect(container.querySelectorAll('.neon-filter-list__item').length).toEqual(longItemList.length);
  });

  it('renders long list no displayCount', async () => {
    const { container, rerender } = harness;
    await rerender({ items: longItemList, modelValue: [] });
    expect(container.querySelectorAll('.neon-filter-list__item').length).toEqual(longItemList.length);
  });

  it('renders single select long list', async () => {
    const { html, rerender } = harness;
    await rerender({ items: longItemList, multiple: false, modelValue: '' });
    expect(html()).toMatchSnapshot();
  });

  it('renders long list, show all', async () => {
    const { html, rerender } = harness;
    await rerender({ items: longItemList, modelValue: [] });
    expect(html()).toMatchSnapshot();
  });

  it('renders default color', () => {
    const { container } = harness;
    expect(container.querySelector('.neon-filter-list--primary')).toBeDefined();
    expect(container.querySelector('.neon-filter-list__item-close.neon-icon--primary')).toBeDefined();
  });

  it('renders color', async () => {
    const { container, rerender } = harness;
    await rerender({ color: NeonFunctionalColor.Brand });
    expect(container.querySelector('.neon-filter-list--brand')).toBeDefined();
    expect(container.querySelector('.neon-filter-list__item-close.neon-icon--brand')).toBeDefined();
  });

  it('renders default size', () => {
    const { container } = harness;
    expect(container.querySelector('.neon-filter-list--m')).toBeDefined();
    expect(container.querySelector('.neon-list--m')).toBeDefined();
  });

  it('renders size', async () => {
    const { container, rerender } = harness;
    await rerender({ size: NeonSize.Large });
    expect(container.querySelector('.neon-filter-list--l')).toBeDefined();
    expect(container.querySelector('.neon-list--l')).toBeDefined();
  });

  it('renders multiple', () => {
    const { container } = harness;
    expect(container.querySelector('.neon-filter-list--multiple')).toBeDefined();
    expect(container.querySelector('.neon-filter-list')?.getAttribute('aria-multiselectable')).toEqual('true');
  });

  it('renders single', async () => {
    const { container, rerender } = harness;
    await rerender({ items, multiple: false, modelValue: '' });
    expect(container.querySelector('.neon-filter-list--multiple')).toBeNull();
    expect(container.querySelector('.neon-filter-list')?.getAttribute('aria-multiselectable')).toEqual('false');
  });

  it('renders active descendant multiple', async () => {
    const { container, rerender } = harness;
    await rerender({ items, modelValue: [items[0].key] });
    expect(container.querySelector('.neon-filter-list')?.getAttribute('aria-activedescendant')).toEqual(items[0].key);
  });

  it('renders active descendant single', async () => {
    const { container, rerender } = harness;
    await rerender({ items, multiple: false, modelValue: items[0].key });
    expect(container.querySelector('.neon-filter-list')?.getAttribute('aria-activedescendant')).toEqual(items[0].key);
  });

  it('renders selected multiple', async () => {
    const { container, rerender } = harness;
    await rerender({ items, modelValue: [items[0].key, items[1].key] });
    const firstEl = container.querySelector('.neon-filter-list__item:first-child.neon-filter-list__item--selected');
    expect(firstEl).toBeDefined();
    expect(firstEl?.getAttribute('aria-selected')).toEqual('true');
    expect(
      container.querySelector('.neon-filter-list__item:nth-child(2).neon-filter-list__item--selected'),
    ).toBeDefined();
  });

  it('renders selected single', async () => {
    const { container, rerender } = harness;
    await rerender({ items, multiple: false, modelValue: items[0].key });
    const el = container.querySelector('.neon-filter-list__item:first-child.neon-filter-list__item--selected');
    expect(el).toBeDefined();
    expect(el?.getAttribute('aria-selected')).toEqual('true');
  });

  it('renders disabled', () => {
    const { container } = harness;
    const el = container.querySelector('.neon-filter-list__item:last-child.neon-filter-list__item--disabled');
    expect(el).toBeDefined();
    expect(el?.getAttribute('tabindex')).toEqual('-1');
  });

  it('renders tabindex', () => {
    const { container } = harness;
    const el = container.querySelector('.neon-filter-list__item:first-child');
    expect(el).toBeDefined();
    expect(el?.getAttribute('tabindex')).toEqual('0');
  });

  it('renders show more label', () => {
    const showMoreLabel = 'xdd';
    const { container } = render(NeonFilterList, {
      props: { items, modelValue: [items[0].key], displayCount: 2, showMoreLabel },
      global: {
        plugins: [router],
      },
    });
    expect(container.querySelector('.neon-filter-list__show-toggle')?.textContent).toEqual(showMoreLabel);
  });

  it('renders show less label', async () => {
    const showLessLabel = 'xdd';
    const showMoreLabel = 'show more';
    const { container } = render(NeonFilterList, {
      props: { items, modelValue: [], displayCount: 2, showMoreLabel, showLessLabel },
      global: {
        plugins: [router],
      },
    });
    const el = container.querySelector('.neon-filter-list__show-toggle') as HTMLElement;
    await fireEvent.click(el);
    expect(container.querySelector('.neon-filter-list__show-toggle')?.textContent).toEqual(showLessLabel);
  });

  it('emits input on click multiple', async () => {
    const { container, emitted, rerender } = harness;
    await rerender({ items, modelValue: [items[1].key] });
    const el = container.querySelector('.neon-filter-list__item:first-child') as HTMLElement;
    await fireEvent.click(el);
    expect(emitted()['update:modelValue'][0]).toEqual([[items[1].key, items[0].key]]);
  });

  it('emits input on keydown enter', async () => {
    const { container, emitted, rerender } = harness;
    await rerender({ items, modelValue: [items[1].key] });
    const el = container.querySelector('.neon-filter-list__item:first-child') as HTMLElement;
    await fireEvent.keyDown(el, { key: 'Enter', code: 'Enter' });
    expect(emitted()['update:modelValue'][0]).toEqual([[items[1].key, items[0].key]]);
  });

  it('emits input on keydown space', async () => {
    const { container, emitted, rerender } = harness;
    await rerender({ items, modelValue: [items[1].key] });
    const el = container.querySelector('.neon-filter-list__item:first-child') as HTMLElement;
    await fireEvent.keyDown(el, { key: 'Space', code: 'Space' });
    expect(emitted()['update:modelValue'][0]).toEqual([[items[1].key, items[0].key]]);
  });

  it('emits input on click multiple deselect', async () => {
    const { container, emitted, rerender } = harness;
    await rerender({ items, modelValue: [items[0].key] });
    const el = container.querySelector('.neon-filter-list__item:first-child') as HTMLElement;
    await fireEvent.click(el);
    expect(emitted()['update:modelValue'][0]).toEqual([[]]);
  });

  it('emits input on click single', async () => {
    const { container, emitted, rerender } = harness;
    await rerender({ items, multiple: false, modelValue: items[1].key });
    const el = container.querySelector('.neon-filter-list__item:first-child') as HTMLElement;
    await fireEvent.click(el);
    expect(emitted()['update:modelValue'][0]).toEqual([items[0].key]);
  });

  it('emits input on click single unselect', async () => {
    const { container, emitted, rerender } = harness;
    await rerender({ items, multiple: false, modelValue: items[0].key });
    const el = container.querySelector('.neon-filter-list__item:first-child') as HTMLElement;
    await fireEvent.click(el);
    expect(emitted()['update:modelValue'][0]).toEqual(['']);
  });

  it('does not emit input on click when disabled', async () => {
    const { container, emitted, rerender } = harness;
    await rerender({ items, modelValue: [items[1].key] });
    const el = container.querySelector('.neon-filter-list__item:last-child') as HTMLElement;
    await fireEvent.click(el);
    expect(emitted()['update:modelValue']).toBeUndefined();
  });
});
