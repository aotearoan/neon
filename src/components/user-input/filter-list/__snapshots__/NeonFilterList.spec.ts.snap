// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NeonFilterList renders long list, show all 1`] = `
<div role="listbox" aria-multiselectable="true" class="neon-filter-list neon-filter-list--low-contrast neon-list--m neon-filter-list--m neon-filter-list--multiple">
  <div tabindex="0" role="option" class="neon-list__item neon-filter-list__item"><span class="neon-filter-list__item-label">Item 1</span> <span class="neon-filter-list__item-count">5,837</span>
    <!---->
  </div>
  <div tabindex="0" role="option" class="neon-list__item neon-filter-list__item"><span class="neon-filter-list__item-label">Item 2</span> <span class="neon-filter-list__item-count">433</span>
    <!---->
  </div>
  <div tabindex="0" role="option" class="neon-list__item neon-filter-list__item"><span class="neon-filter-list__item-label">Item 3</span> <span class="neon-filter-list__item-count">327</span>
    <!---->
  </div>
  <div tabindex="0" role="option" class="neon-list__item neon-filter-list__item"><span class="neon-filter-list__item-label">Item 4</span> <span class="neon-filter-list__item-count">100</span>
    <!---->
  </div>
  <div tabindex="0" role="option" class="neon-list__item neon-filter-list__item"><span class="neon-filter-list__item-label">Item 5</span> <span class="neon-filter-list__item-count">5,837</span>
    <!---->
  </div>
  <div tabindex="0" role="option" class="neon-list__item neon-filter-list__item"><span class="neon-filter-list__item-label">Item 6</span> <span class="neon-filter-list__item-count">433</span>
    <!---->
  </div>
  <div tabindex="0" role="option" class="neon-list__item neon-filter-list__item"><span class="neon-filter-list__item-label">Item 7</span> <span class="neon-filter-list__item-count">327</span>
    <!---->
  </div>
  <div tabindex="0" role="option" class="neon-list__item neon-filter-list__item"><span class="neon-filter-list__item-label">Item 8</span> <span class="neon-filter-list__item-count">100</span>
    <!---->
  </div>
  <!---->
</div>
`;

exports[`NeonFilterList renders multiselect short list 1`] = `
<div role="listbox" aria-multiselectable="true" class="neon-filter-list neon-filter-list--low-contrast neon-list--m neon-filter-list--m neon-filter-list--multiple">
  <div tabindex="0" role="option" class="neon-list__item neon-filter-list__item"><span class="neon-filter-list__item-label">Item 1</span> <span class="neon-filter-list__item-count">5,837</span>
    <!---->
  </div>
  <div tabindex="0" role="option" class="neon-list__item neon-filter-list__item"><span class="neon-filter-list__item-label">Item 2</span> <span class="neon-filter-list__item-count">433</span>
    <!---->
  </div>
  <div tabindex="0" role="option" class="neon-list__item neon-filter-list__item"><span class="neon-filter-list__item-label">Item 3</span> <span class="neon-filter-list__item-count">327</span>
    <!---->
  </div>
  <div tabindex="-1" role="option" class="neon-list__item neon-filter-list__item neon-filter-list__item--disabled"><span class="neon-filter-list__item-label">Disabled item</span> <span class="neon-filter-list__item-count">100</span>
    <!---->
  </div>
  <!---->
</div>
`;

exports[`NeonFilterList renders single select long list 1`] = `
<div role="listbox" aria-activedescendant="" class="neon-filter-list neon-filter-list--low-contrast neon-list--m neon-filter-list--m">
  <div tabindex="0" role="option" class="neon-list__item neon-filter-list__item"><span class="neon-filter-list__item-label">Item 1</span> <span class="neon-filter-list__item-count">5,837</span>
    <!---->
  </div>
  <div tabindex="0" role="option" class="neon-list__item neon-filter-list__item"><span class="neon-filter-list__item-label">Item 2</span> <span class="neon-filter-list__item-count">433</span>
    <!---->
  </div>
  <div tabindex="0" role="option" class="neon-list__item neon-filter-list__item"><span class="neon-filter-list__item-label">Item 3</span> <span class="neon-filter-list__item-count">327</span>
    <!---->
  </div>
  <div tabindex="0" role="option" class="neon-list__item neon-filter-list__item"><span class="neon-filter-list__item-label">Item 4</span> <span class="neon-filter-list__item-count">100</span>
    <!---->
  </div>
  <div tabindex="0" role="option" class="neon-list__item neon-filter-list__item"><span class="neon-filter-list__item-label">Item 5</span> <span class="neon-filter-list__item-count">5,837</span>
    <!---->
  </div>
  <div tabindex="0" role="option" class="neon-list__item neon-filter-list__item"><span class="neon-filter-list__item-label">Item 6</span> <span class="neon-filter-list__item-count">433</span>
    <!---->
  </div>
  <div tabindex="0" role="option" class="neon-list__item neon-filter-list__item"><span class="neon-filter-list__item-label">Item 7</span> <span class="neon-filter-list__item-count">327</span>
    <!---->
  </div>
  <div tabindex="0" role="option" class="neon-list__item neon-filter-list__item"><span class="neon-filter-list__item-label">Item 8</span> <span class="neon-filter-list__item-count">100</span>
    <!---->
  </div>
  <!---->
</div>
`;
